astropy/table/table.py

```diff
@@ -3644,12 +3648,33 @@ def _convert_col_for_table(self, col):
             qcol = q_cls(col.data, col.unit, copy=False)
             qcol.info = col.info
             qcol.info.indices = col.info.indices
+            if np.issubdtype(getattr(col, 'dtype', None), np.int):
```

Comment:
Change `np.int` to `np.integer` (which is the actual type class, while `np.int is int`).